src/error.rs

```diff
@@ -74,6 +74,9 @@ pub enum ApiErrorKind {
 
     #[fail(display = "{}", _0)]
     Validation(#[cause] ValidationError),
+
+    #[fail(display = "Invalid submission: {}", _0)]
+    Invalid(String),
```

Comment:
I imagine there's multiple ways for a request to be invalid. Maybe renaming this to `InvalidSubmission` would make sense?

Reply:
Even better! It's unused. Dropping it.